### PR TITLE
Property resolution in activity mapping

### PR DESCRIPTION
**Pull request by vijaynalawade**
Changes to support property resolution

### CODE REVIEW

File: core/data/path.go [hunk 1]
```diff
@@ -41,6 +42,15 @@ func GetAttrPath(inAttrName string) (attrName string, attrPath string, pathType
 				attrPath = inAttrName[idx+2:]
 			}
 		}
+	} else if inAttrName[0] == '$' && inAttrName[1] == '{' {
```
Comment (torresashjian) [inline core/data/path.go]: We should use strings.HasPrefix() function here. That will protect against inAttrName smaller than 2 in len().

In fact, if you check for prefixes '${property.' || '${env.' you remove quite a bit of code there, and will not affect any other part of the if statement.

} else if strings.HasPrefix('${property.') || strings.HasPrefix('${env.'){

File: core/property/registry.go [hunk 2]
```diff
@@ -84,10 +84,13 @@ func getValueFromResolver(prop interface{}) (interface{}, bool) {
 // expressions, the resolved value is returned and the boolean is true.
 // Otherwise the returned value will be nil and the boolean will be false.
 
-func Resolve(name string) (interface{}, bool) {
+func Resolve(value interface{}) (interface{}, bool) {
```
Comment (torresashjian) [inline core/property/registry.go]: why are we resolving non string? Sorry, a method with input and output interface type gives no information to the user of what is valid whatsoever.

Comment (vijaynalawade) [inline core/property/registry.go]: Yes. Will revert that change as that it is not needed.

File: core/data/path.go (hunk 1 above)
Comment (vijaynalawade) [inline core/data/path.go]: We will be adding ${activity.*} and ${trigger.*} and so on. Hence, decided to go with this approach.

File: core/data/path_test.go [hunk 3]
```diff
@@ -35,4 +35,12 @@ func TestGetAttrPath(t *testing.T) {
 	a = "{T1.v}"
 	name, path, pt = GetAttrPath(a)
 	fmt.Printf("Name: %s, Path: %s, PathType: %d\n", name, path, pt)
+	
+	a = "${property.Prop1}"
```
Comment (torresashjian) [inline core/data/path_test.go]: I know this was here already, but what is this test doing? shouldn't we assert?

**Pull request merged**